torch/_inductor/codegen/wrapper.py

```diff
@@ -141,7 +141,14 @@ def codegen(self, code: IndentedBuffer, device_cm_stack: contextlib.ExitStack):
         if V.graph.cpp_wrapper:
             code.writeline("\n")
             if self.first_time:
-                code.writeline(f"at::cuda::CUDAGuard device_guard({self.device_idx});")
+                if torch.version.hip is None:
+                    code.writeline(
+                        f"at::cuda::CUDAGuard device_guard({self.device_idx});"
+                    )
+                else:
+                    code.writeline(
+                        f"at::hip::HIPGuard device_guard({self.device_idx});"
+                    )
```

Comment:
@jataylo I'll need to dig further into the history to figure out why, but hipify actually has the following mapping for `CUDAGuard`:
`("CUDAGuard", ("HIPGuardMasqueradingAsCUDA", API_PYTORCH)),`
Did you try that already and run into some issues?

Reply:
Initially I had manually tried to use HIPGuardMasqueradingAsCUDA but ran into some issues, likely a mistake on my part. With the hipify integration this works as expected.